### PR TITLE
feat: CP-9472 add duplicated account dialog

### PR DESCRIPTION
**Pull request by vvava**
## Description

We want to notify the user when a private key is being imported and that is already exist as a primary wallet.
Also we want to notify the users in a sooner step when they want to add a ledger wallet which is already lives in the extension.

## Changes

New dialog to notify the users about duplicated account.
Modified import ledger handler so it can be used as a monitoring function to get the wallet is exist.
Modified UI flow to notify the users the ledger wallet already exists before we  moved them forward to the naming phase.

## Testing

Go to the Accounts page -> try to add the existing mnemonic wallet by importing the private key
and check you cannot add the same ledger (with BIP44 and Ledger Live) multiple times.

## Screenshots:

image

https://github.com/user-attachments/assets/72956a20-de26-41b8-9492-25ba796c729c

## Checklist for the author

Tick each of them when done or if not applicable.

- [x] I've covered new/modified business logic with Jest test cases.
- [x] I've tested the changes myself before sending it to code review and QA.

### CODE REVIEW

File: src/background/services/wallet/handlers/importLedger.ts [hunk 1]
```diff
@@ -68,6 +70,16 @@ export class ImportLedgerHandler implements HandlerType {
       };
     }
 
+    if (onlyCheckWalletIsExist) {
+      return {
+        ...request,
+        result: {
+          id: '0',
+          type: secretType,
+        },
```
Comment (gergelylovas) [inline src/background/services/wallet/handlers/importLedger.ts:79]: Let's try to preserve the same response type as the success message normally has.
```suggestion
        result: {
          id: '0',
          name,
          type: secretType,
        },
```

File: src/background/services/wallet/handlers/models.ts [hunk 2]
```diff
@@ -23,6 +23,7 @@ export type ImportLedgerWalletParams = {
   pubKeys?: PubKeyType[];
   secretType: SecretType.Ledger | SecretType.LedgerLive;
   name?: string;
+  onlyCheckWalletIsExist?: boolean;
```
Comment (gergelylovas) [inline src/background/services/wallet/handlers/models.ts:26]: Should we call this `dryRun` instead? 
```suggestion
  dryRun?: boolean;
```

File: src/pages/ImportPrivateKey/ImportPrivateKey.tsx [hunk 3]
```diff
@@ -49,14 +51,38 @@ export function ImportPrivateKey() {
...
+        ({ addressC }) => addressC.toLowerCase() === address.toLowerCase()
+      );
+      if (findAccount) {
+        setIsKnownAccount(true);
+      }
+    },
+    [allAccounts]
+  );
+
+  console.log('allAccounts: ', allAccounts);
```
Comment (gergelylovas) [inline src/pages/ImportPrivateKey/ImportPrivateKey.tsx:73]: Please remove

File: src/components/ledger/LedgerConnector.tsx [hunk 4]
```diff
@@ -48,11 +50,13 @@ export interface LedgerConnectorData {
 interface LedgerConnectorProps {
   onSuccess: (data: LedgerConnectorData) => void;
   onTroubleshoot: () => void;
+  checkWalletIsExist?: boolean;
```
Comment (gergelylovas) [inline src/components/ledger/LedgerConnector.tsx:53]:
```suggestion
  checkWalletExists?: boolean;
```

**Review by gergelylovas — CHANGES_REQUESTED** [review]

**Review by bferenc — CHANGES_REQUESTED** [review]
Works great! 👍 
Can we standardize the names though? Currently there are multiple cases like "checkWalletExists", "isLedgerExists", "checkLedgerWalletIsExist"

**Review by bferenc — APPROVED** [review]
Tested as well ✅

File: src/pages/ImportPrivateKey/ImportPrivateKey.tsx [hunk 5]
```diff
@@ -49,14 +51,36 @@ export function ImportPrivateKey() {
   const balance = useBalanceTotalInCurrency(derivedAddresses as Account);
   const { isImporting: isImportLoading, importPrivateKey } =
     usePrivateKeyImport();
+  const { selectAccount } = useAccountsContext();
   const history = useHistory();
+  const { allAccounts } = useAccountsContext();
+  const [isKnownAccount, setIsKnownAccount] = useState(false);
+  const [isDuplicatedAccountDialogOpen, setIsDuplicatedDialogDialogOpen] =
```
Comment (meeh0w) [inline src/pages/ImportPrivateKey/ImportPrivateKey.tsx:58]:
```suggestion
  const [isDuplicatedAccountDialogOpen, setIsDuplicatedAccountDialogOpen] =
```

File: src/pages/ImportPrivateKey/ImportPrivateKey.tsx [hunk 6]
```diff
@@ -49,14 +51,36 @@ export function ImportPrivateKey() {
...
+    useState(false);
+
+  const checkAccount = useCallback(
```
Comment (meeh0w) [inline src/pages/ImportPrivateKey/ImportPrivateKey.tsx:61]:
```suggestion
  const checkIfAccountExists = useCallback(
```

File: src/components/ledger/LedgerConnector.tsx [hunk 7]
```diff
@@ -110,12 +116,42 @@ export function LedgerConnector({
...
+          pubKeys: publicKeys,
+          secretType:
+            derivationPath === DerivationPath.BIP44
+              ? SecretType.Ledger
+              : SecretType.LedgerLive,
+          dryRun: true,
+        });
+      } catch (e) {
+        setIsLedgerExistsError(true);
+        throw new Error('wallet already exists');
```
Comment (meeh0w) [inline src/components/ledger/LedgerConnector.tsx:135]: We should check the error message - there are other reasons `importLedger(...)` could fail.

File: src/components/ledger/LedgerConnector.tsx [hunk 8]
```diff
@@ -48,11 +50,13 @@ export interface LedgerConnectorData {
 interface LedgerConnectorProps {
   onSuccess: (data: LedgerConnectorData) => void;
   onTroubleshoot: () => void;
+  isWalletExist?: boolean;
 }
 
 export function LedgerConnector({
   onSuccess,
   onTroubleshoot,
+  isWalletExist,
```
Comment (meeh0w) [inline src/components/ledger/LedgerConnector.tsx:59]: If I understand correctly, this prop tells `LedgerConnector` to verify if the secret already exists, right?

If so, I think we should rename it. Right now it sounds more as a statement than a question/request to check for something.

Maybe: `checkIfWalletExists` or `withWalletDeduplicationWarning` ?

**Review by meeh0w — CHANGES_REQUESTED** [review]

**Review by meeh0w — APPROVED** [review]

**Review by bferenc — APPROVED** [review]

**Comment by meeh0w** [comment]
(I also tested, btw.)

**Review by bferenc — APPROVED** [review]

**Review by meeh0w — APPROVED** [review]

**Pull request merged**